internal/pkg/pluginmanager/downloader_test.go

```diff
@@ -1,29 +1,55 @@
 package pluginmanager
 
 import (
+	"errors"
+	"fmt"
+	"log"
 	"os"
 	"path/filepath"
 	"testing"
 
+	"github.com/go-resty/resty/v2"
 	"github.com/stretchr/testify/assert"
 )
 
-func TestDownload(t *testing.T) {
-	os.Remove(filepath.Join(".", "argocdapp_0.0.1-rc1.so"))
+func mockPlugGetter(reqClient *resty.Client, url, plugName string) error {
+	return nil
+}
+
+func mockPlugNotFoundGetter(reqClient *resty.Client, url, plugName string) error {
+	return errors.New("Plug Not Exist")
```

Comment:
Please note that when downloading fails, it returns `fmt.Errorf("downloading plugin %s from %s status code %d", plugName, url, response.StatusCode())` NOT `errors.New("Plug Not Exist")`

Reply:
Titlecase isn't necessary for error messages.

Error messages should not be a full sentence, so the first character should not be uppercase.